scripts/convert_original_stable_diffusion_to_diffusers.py

```diff
@@ -855,11 +862,17 @@ def convert_open_clip_checkpoint(checkpoint):
     image_size = args.image_size
     prediction_type = args.prediction_type
 
-    if args.device is None:
-        device = "cuda" if torch.cuda.is_available() else "cpu"
-        checkpoint = torch.load(args.checkpoint_path, map_location=device)
+    if args.from_safetensors:
+        checkpoint = {}
+        with safe_open(args.checkpoint_path, framework="pt", device="cpu") as f:
+            for key in f.keys():
+                checkpoint[key] = f.get_tensor(key)
```

Comment:
Looks good. Should we also try to detect if the weights are `safetensors` format? Maybe using the file extension.

Reply:
Many people save safetensors with .ckpt so for now, I'm not too keen on adding too much magic there

Reply:
See: https://huggingface.co/spaces/diffusers/sd-to-diffusers/discussions/1